docs/unity/part-1.md

```diff
@@ -553,7 +561,17 @@ public class RemotePlayer : MonoBehaviour
         canvas.worldCamera = Camera.main;
 
         // Get the username from the PlayerComponent for this object and set it in the UI
-        PlayerComponent playerComp = PlayerComponent.FilterByEntityId(EntityId);
+        PlayerComponent? playerComp = PlayerComponent.FilterByEntityId(EntityId).FirstOrDefault();
+        if (playerComp is null)
+        {
+            string inputUsername = UsernameElement.Text;
+            Debug.Log($"PlayerComponent not found - Creating a new player ({inputUsername})");
+            Reducer.CreatePlayer(inputUsername);
+            
+            // Try again, optimistically assuming success for simplicity
+            PlayerComponent? playerComp = PlayerComponent.FilterByEntityId(EntityId).FirstOrDefault();
+        }
```

Comment:
was this change necessary for some reason? It seems like a bug to have somehow created a `RemotePlayer` with an invalid `EntityId`?

Reply:
> was this change necessary for some reason? It seems like a bug to have somehow created a `RemotePlayer` with an invalid `EntityId`?

You're right! This change was unnecessarily, forgetting this was a _remote_ player. Upon this section revert, it may be worth renaming `playerComp` to `remotePlayerComp`? if I can miss it, others may too.

Reply:
![image](https://github.com/clockworklabs/spacetime-docs/assets/8840024/bd1ba305-b627-4a44-96c7-a413eb4293b7)